allure-model/src/main/java/io/qameta/allure/model/TestResult.java

```diff
@@ -0,0 +1,441 @@
+        if (values != null) {
+            for (Label value: values) {
+                getLabels().add(value);
+            }
+        }
+        return this;
+    }
+
+    public TestResult setLabels(final Collection<Label> values) {
+        if (values != null) {
+            getLabels().addAll(values);
+        }
+        return this;
+    }
+
+    public List<Link> getLinks() {
+        if (links == null) {
+            links = new ArrayList<>();
+        }
+        return links;
+    }
+
+    public TestResult setLinks(final List<Link> links) {
+        this.links = links;
+        return this;
+    }
+
+    public TestResult setLinks(final Link... values) {
+        if (values != null) {
+            for (Link value: values) {
+                getLinks().add(value);
+            }
+        }
+        return this;
+    }
+
+    public TestResult setLinks(final Collection<Link> values) {
+        if (values != null) {
+            getLinks().addAll(values);
+        }
+        return this;
+    }
+
+    /**
+     * @deprecated use set method
+     */
+    @Deprecated
+    public TestResult withUuid(final String value) {
+        return setUuid(value);
+    }
+
+    /**
+     * @deprecated use set method
+     */
+    @Deprecated
+    public TestResult withHistoryId(final String value) {
+        return setHistoryId(value);
+    }
+
+    /**
+     * @deprecated use set method
+     */
+    @Deprecated
+    public TestResult withTestCaseId(final String value) {
+        return setTestCaseId(value);
+    }
+
+    /**
+     * @deprecated use set method
+     */
+    @Deprecated
+    public TestResult withRerunOf(final String value) {
+        return setRerunOf(value);
+    }
+
+    /**
+     * @deprecated use set method
+     */
+    @Deprecated
+    public TestResult withFullName(final String value) {
+        return setFullName(value);
+    }
+
+    /**
+     * @deprecated use set method
+     */
+    @Deprecated
+    public TestResult withLabels(final Label... values) {
+        return setLabels(values);
+    }
+
+    /**
+     * @deprecated use set method
+     */
+    @Deprecated
+    public TestResult withLabels(final Collection<Label> values) {
+        return setLabels(values);
+    }
+
+    /**
+     * @deprecated use set method
+     */
+    @Deprecated
+    public TestResult withLabels(final List<Label> labels) {
+        return setLabels(labels);
+    }
+
+    /**
+     * @deprecated use set method
+     */
+    @Deprecated
+    public TestResult withLinks(final Link... values) {
+        return setLinks(values);
+    }
+
+    /**
+     * @deprecated use set method
+     */
+    @Deprecated
+    public TestResult withLinks(final Collection<Link> values) {
+        return setLinks(values);
+    }
+
+    /**
+     * @deprecated use set method
+     */
+    @Deprecated
+    public TestResult withLinks(final List<Link> links) {
+        return setLinks(links);
+    }
+
+    @Override
+    public TestResult withName(final String value) {
```

Comment:
add deprecation. Also comment should be `scheduled to remove in 3.0 release. Use setter method instead`